nym-vpn-core/crates/nym-vpn-lib/src/platform/error.rs

```diff
@@ -21,41 +21,8 @@ pub enum VpnError {
     #[error("{details}")]
     InvalidStateError { details: String },
 
-    #[error("{0}")]
-    Account(AccountError),
-}
-
-#[derive(thiserror::Error, uniffi::Error, Debug, Clone, PartialEq)]
-pub enum AccountError {
-    #[error("account state is ready to connect")]
-    Ready,
-    #[error("no account stored")]
-    NoAccountStored,
-    #[error("account not active")]
-    AccountNotActive,
-    #[error("no active subscription")]
-    NoActiveSubscription,
-    #[error("device not registered")]
-    DeviceNotRegistered,
-    #[error("device not active")]
-    DeviceNotActive,
-}
-
-impl From<nym_vpn_account_controller::ReadyToConnect> for AccountError {
-    fn from(value: nym_vpn_account_controller::ReadyToConnect) -> Self {
-        match value {
-            nym_vpn_account_controller::ReadyToConnect::Ready => Self::Ready,
-            nym_vpn_account_controller::ReadyToConnect::NoMnemonicStored => Self::NoAccountStored,
-            nym_vpn_account_controller::ReadyToConnect::AccountNotActive => Self::AccountNotActive,
-            nym_vpn_account_controller::ReadyToConnect::NoActiveSubscription => {
-                Self::NoActiveSubscription
-            }
-            nym_vpn_account_controller::ReadyToConnect::DeviceNotRegistered => {
-                Self::DeviceNotRegistered
-            }
-            nym_vpn_account_controller::ReadyToConnect::DeviceNotActive => Self::DeviceNotActive,
-        }
-    }
+    #[error("{details}")]
+    Account { details: String },
```

Comment:
This is unfortunate, but oh well you gotta do what you gotta do. Maybe we can return an integer instead so that at least you can localize the response?